# mayhem_program::instructions::pump_amm_interface::sell_pump_swap::handler  (yours 12104 B at .text+0x1a520; original 17312 B at .text+0x17548)
# distance 3013 instructions; evidence: called-by mayhem_program::__private::__global::sell_pump_swap#777f; same-callees 18/19
--- original
+++ yours
@@ -1,2057 +1,1336 @@
-mov64 r6, r5
-mov64 r7, r4
-stxdw [r10-1744], r3
-stxdw [r10-1736], r2
-mov64 r8, r1
-ldxdw r9, [r8+8]
+stxdw [r10-2760], r5
+mov64 r6, r4
+stxdw [r10-2736], r3
+stxdw [r10-2744], r2
+ldxdw r9, [r1+8]
+ldxdw r2, [r9+104]
+ldxdw r4, [r2]
+ldxdw r2, [r4]
+mov64 r3, 1
+stxdw [r10-2752], r3
+lddw r3, 0x8481abfe57889b06
+mov64 r7, 1
+jeq r2, r3, +0x78
+mov64 r7, 0
+ldxdw r3, [r4+8]
+lddw r0, 0x35c01846637f68fb
+mov64 r5, 1
+jeq r3, r0, +0xa8
+mov64 r5, 0
+and64 r7, r5
+ldxdw r5, [r4+16]
+lddw r8, 0x553beb1adc39c4da
+mov64 r0, 1
+jeq r5, r8, +0xe0
+mov64 r0, 0
+and64 r7, r0
+ldxdw r4, [r4+24]
+lddw r8, 0x100000000f0a098
+mov64 r0, 1
+jeq r4, r8, +0x118
+mov64 r0, 0
+and64 r7, r0
+and64 r7, 1
+jne r7, 0, +0x150
+jne r2, 0, +0x1f8
+jne r3, 0, +0x1f8
+jne r5, 0, +0x1f8
+jne r4, 0, +0x1f8
 ldxdw r2, [r9+8]
-ldxdw r1, [r9+16]
-add64 r1, 8
-add64 r2, 8
-call orig_fn_2dbf8
-jne r0, 2, +0xad8
-stxdw [r10-1760], r8
-stxdw [r10-1752], r6
+ldxb r2, [r2+253]
+jeq r2, 0, +0x220
+mov64 r1, r10
+add64 r1, -2656
+lddw r2, .rodata[00000000]
+call mayhem_program::MayhemError::name#1311
+mov64 r1, r10
+add64 r1, -2632
+mov64 r2, 0
+call <T as alloc::string::SpecToString>::spec_to_string#de30
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-2680], r1
+stb [r10-2608], 2
+stw [r10-2664], 294
+stdw [r10-2672], 62
+stdw [r10-2688], 0
+stw [r10-2536], 6000
+ja +0x608
+mov64 r2, 0
+stxdw [r10-2752], r2
+ldxdw r2, [r9+8]
+ldxb r2, [r2+253]
+jne r2, 0, +0x168
+stxdw [r10-2776], r1
+mov64 r1, r10
+add64 r1, -2688
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-2688]
+jne r1, 0, +0x2d18
+ldxdw r1, [r10-2648]
+ldxdw r2, [r9+16]
+ldxdw r3, [r2+88]
+jeq r3, 0, +0x370
+ldxdw r4, [r9+8]
+ldxdw r8, [r4+237]
+jeq r8, 0, +0x370
+mov64 r0, 1
+mov64 r4, 1
+jslt r8, 0, +0x530
+mov64 r4, 0
+mov64 r5, r3
+add64 r5, r8
+jsge r5, r3, +0x548
+xor64 r4, r0
+and64 r4, 1
+jne r4, 0, +0x568
+jsge r1, r5, +0x370
+mov64 r1, r10
+add64 r1, -2656
+lddw r2, .rodata[0f000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#61b2
+mov64 r1, r10
+add64 r1, -2632
+mov64 r2, 15
+call <T as alloc::string::SpecToString>::spec_to_string#c4bf
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-2680], r1
+stb [r10-2608], 2
+stw [r10-2664], 298
+stdw [r10-2672], 62
+stdw [r10-2688], 0
+stw [r10-2536], 6015
+ja +0x608
+stxdw [r2+88], r1
 ldxdw r1, [r9+120]
 ldxdw r3, [r1+104]
-ldxdw r1, [r9+88]
-ldxdw r2, [r1+232]
-mov64 r8, r3
-add64 r8, r2
-mov64 r2, 1
-jlt r8, r3, +0xb8
+ldxdw r2, [r9+88]
+ldxdw r4, [r2+232]
+mov64 r1, r3
+add64 r1, r4
+mov64 r4, 1
+jlt r1, r3, +0x3c0
+mov64 r4, 0
+ldxdw r3, [r2+240]
+mov64 r2, r3
+add64 r2, r4
+xor64 r3, -1
+and64 r3, r2
+jslt r3, 0, +0x9b8
+jne r2, 0, +0x1f58
+stxdw [r10-2768], r1
+mov64 r1, r10
+add64 r1, -2688
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-2688]
+jeq r1, 1, +0x2d78
+ldxdw r2, [r10-2648]
+ldxdw r1, [r9+16]
+ldxdw r3, [r1+80]
+jsge r2, r3, +0x580
+ldxb r3, [r1+104]
+ldxdw r2, [r10-2768]
+jeq r3, 1, +0x630
+mov64 r3, 144
+ldxdw r4, [r10-2752]
+jne r4, 0, +0x480
+mov64 r3, 192
+ldxdw r4, [r9+8]
+add64 r4, r3
+ldxdw r3, [r4]
+jge r2, r3, +0x628
+mov64 r1, r10
+add64 r1, -2656
+lddw r2, .rodata[030000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#61b2
+mov64 r1, r10
+add64 r1, -2632
+mov64 r2, 3
+call <T as alloc::string::SpecToString>::spec_to_string#c4bf
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-2680], r1
+stb [r10-2608], 2
+stw [r10-2664], 303
+stdw [r10-2672], 62
+stdw [r10-2688], 0
+stw [r10-2536], 6003
+ja +0x608
+mov64 r5, r3
+add64 r5, r8
+jslt r5, r3, +0x2c0
+mov64 r0, 0
+xor64 r4, r0
+and64 r4, 1
+jeq r4, 0, +0x2d8
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:298:41)
+call core::panicking::panic_const::panic_const_add_overflow#3493
+mov64 r1, r10
+add64 r1, -2656
+lddw r2, .rodata[04000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#61b2
+mov64 r1, r10
+add64 r1, -2632
+mov64 r2, 4
+call <T as alloc::string::SpecToString>::spec_to_string#c4bf
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-2680], r1
+stb [r10-2608], 2
+stw [r10-2664], 301
+stdw [r10-2672], 62
+stdw [r10-2688], 0
+stw [r10-2536], 6004
+mov64 r1, r10
+add64 r1, -2688
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+exit
+stb [r1+104], 1
+ldxdw r1, [r9+112]
+ldxdw r1, [r1+104]
+stxdw [r10-2784], r1
+jeq r1, 0, +0x9b8
+stxdw [r10-2792], r9
+and64 r6, 65535
+ldxdw r1, [r10-2736]
 mov64 r2, 0
-ldxdw r1, [r1+240]
+mov64 r3, r6
+mov64 r4, 0
+call __multi3
+mov64 r9, r0
+mov64 r8, r1
+ldxdw r1, [r10-2744]
+mov64 r2, 0
+mov64 r3, r6
+mov64 r4, 0
+call __multi3
+mov64 r2, r1
+add64 r2, r9
+mov64 r4, 1
+mov64 r3, 1
+jne r8, 0, +0x710
+mov64 r3, 0
+jge r2, r1, +0x718
+or64 r3, r4
+jne r3, 1, +0x730
+ja +0x9b8
+jlt r2, r1, +0x6f8
+mov64 r4, 0
+or64 r3, r4
+jeq r3, 1, +0x9b8
+mov64 r1, r0
+mov64 r3, 10000
+mov64 r4, 0
+call __udivti3
+mov64 r9, r0
 mov64 r6, r1
-add64 r6, r2
-xor64 r1, -1
-and64 r1, r6
-jslt r1, 0, +0x360
-jne r6, 0, +0x360
-ldxdw r3, [r9+8]
-ldxdw r1, [r9+16]
-ldxdw r2, [r9+104]
-ldxdw r2, [r2]
-ldxdw r4, [r2+24]
-stxdw [r10-1704], r4
-ldxdw r4, [r2+16]
-stxdw [r10-1712], r4
-ldxdw r4, [r2+8]
-stxdw [r10-1720], r4
-ldxdw r2, [r2]
-stxdw [r10-1728], r2
-add64 r1, 8
-add64 r3, 8
-mov64 r4, r10
-add64 r4, -1728
-mov64 r2, r8
-call orig_fn_2e0a0
-jne r0, 2, +0xad8
-mov64 r1, 2
-stxdw [r10-1776], r1
-stxdw [r10-1784], r9
-ldxdw r1, [r9+112]
-ldxdw r9, [r1+104]
-jeq r9, 0, +0x978
-mov64 r1, r8
-mov64 r2, r6
+mov64 r1, 1
+mov64 r8, 1
+ldxdw r2, [r10-2744]
+jge r2, r9, +0x960
+mov64 r8, 0
+ldxdw r2, [r10-2768]
+ldxdw r3, [r10-2736]
+jlt r3, r6, +0x978
+ldxdw r3, [r10-2736]
+jeq r3, r6, +0x7b8
+mov64 r8, r1
+mov64 r1, r2
+mov64 r2, 0
 lddw r3, 0x38d7ea4c68000
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
 mov64 r1, r0
-stxdw [r10-1800], r9
+ldxdw r3, [r10-2784]
+mov64 r4, 0
+call __udivti3
+and64 r8, 1
+jeq r8, 0, +0x9b8
+ldxdw r5, [r10-2744]
+mov64 r4, r5
+add64 r4, r9
+mov64 r3, 1
+mov64 r2, 1
+jlt r4, r5, +0x858
+mov64 r2, 0
+ldxdw r5, [r10-2736]
+mov64 r4, r5
+add64 r4, r6
+add64 r4, r2
+jlt r4, r5, +0x998
+mov64 r3, 0
+ldxdw r5, [r10-2736]
+jne r4, r5, +0x9a8
+jeq r2, 1, +0x9b8
 mov64 r3, r9
-mov64 r4, 0
-call __udivti3
-stxdw [r10-1792], r0
-stxdw [r10-1768], r1
-and64 r7, 65535
-ldxdw r1, [r10-1744]
+mov64 r2, 1
+mov64 r4, 1
+ldxdw r5, [r10-2744]
+jlt r5, r3, +0x8d0
+mov64 r4, 0
+ldxdw r5, [r10-2736]
+sub64 r5, r6
+sub64 r5, r4
+ldxdw r8, [r10-2744]
+sub64 r8, r9
+mov64 r4, 1
+jlt r0, r8, +0x9c8
+mov64 r4, 0
+mov64 r8, 1
+jge r1, r5, +0x9d8
+jeq r1, r5, +0x9e8
+mov64 r4, r8
+ldxdw r5, [r10-2744]
+add64 r9, r5
+jge r9, r3, +0xa00
+and64 r4, 1
+jeq r4, 0, +0xa18
+ja +0xa78
+ldxdw r2, [r10-2768]
+ldxdw r3, [r10-2736]
+jge r3, r6, +0x7a0
+mov64 r1, 0
+ldxdw r3, [r10-2736]
+jne r3, r6, +0x7b0
+ja +0x7b8
+ldxdw r5, [r10-2736]
+jeq r4, r5, +0x898
+mov64 r2, r3
+jne r2, 1, +0x8a0
+call mayhem_program::instructions::pump_interface::buy::math#3277
+ja +0x620
+mov64 r8, 1
+jlt r1, r5, +0x920
+mov64 r8, 0
+jne r1, r5, +0x928
+ldxdw r5, [r10-2744]
+add64 r9, r5
+jlt r9, r3, +0x948
 mov64 r2, 0
-mov64 r3, r7
-mov64 r4, 0
-call __multi3
-mov64 r6, r0
-mov64 r9, r1
-ldxdw r1, [r10-1736]
+and64 r4, 1
+jne r4, 0, +0xa78
+ldxdw r3, [r10-2736]
+add64 r6, r3
+add64 r6, r2
+mov64 r3, 1
+mov64 r2, 1
+jle r0, r9, +0xa90
 mov64 r2, 0
-mov64 r3, r7
+jgt r1, r6, +0xa98
+jeq r1, r6, +0xaa8
+mov64 r2, r3
+and64 r2, 1
+jne r2, 0, +0xab8
+mov64 r1, 12
+call mayhem_program::instructions::pump_interface::sell::<impl core::convert::From<mayhem_program::instructions::pump_interface::sell::SellError> for anchor_lang::error::Error>::from#c145
+ja +0x620
+jle r1, r6, +0xa58
+mov64 r3, 0
+jne r1, r6, +0xa60
+and64 r2, 1
+jeq r2, 0, +0xa78
+ldxdw r1, [r10-2760]
+and64 r1, 65535
+jgt r1, 10000, +0x2ed0
+ldxdw r9, [r10-2792]
+ldxdw r1, [r9+32]
+ldxdw r6, [r1+104]
+mov64 r1, r6
+mov64 r2, 0
+ldxdw r3, [r10-2768]
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
-add64 r2, r6
+mov64 r3, r6
+ldxdw r1, [r10-2784]
+add64 r3, r1
 mov64 r4, 1
-mov64 r3, 1
-jne r9, 0, +0x2c0
-mov64 r3, 0
-jlt r2, r1, +0x2d0
-mov64 r4, 0
-or64 r3, r4
-and64 r3, 1
-ldxdw r6, [r10-1784]
-ldxdw r7, [r10-1744]
-jne r3, 0, +0x978
+stxdw [r10-2944], r6
+jlt r3, r6, +0xb50
+mov64 r4, 0
+mov64 r1, r0
+call __udivti3
+mov64 r6, r0
+jne r1, 0, +0x2dd8
+mov64 r2, 112
+ldxdw r1, [r10-2768]
+ldxdw r3, [r10-2752]
+jne r3, 0, +0xb98
+mov64 r2, 168
+mov64 r3, 10000
+ldxdw r4, [r9+8]
+add64 r4, r2
+ldxdw r2, [r4]
+mov64 r4, r2
+jlt r2, r6, +0xbd0
+mov64 r4, r6
+ldxdw r5, [r10-2760]
+sub64 r3, r5
+jeq r2, 0, +0xbf0
+mov64 r6, r4
+and64 r3, 65535
+mov64 r2, 0
+mov64 r4, 0
+call __multi3
+mov64 r2, r1
 mov64 r1, r0
 mov64 r3, 10000
 mov64 r4, 0
 call __udivti3
-mov64 r3, 1
+jlt r6, r0, +0xc48
+mov64 r6, r0
+mov64 r1, r6
+mov64 r2, 0
+ldxdw r3, [r10-2784]
+mov64 r4, 0
+call __multi3
+mov64 r2, r1
+mov64 r4, 1
+ldxdw r3, [r10-2768]
+jlt r3, r6, +0xc98
+mov64 r4, 0
+jlt r3, r6, +0x2ee8
+jeq r3, r6, +0x2e28
+sub64 r3, r6
+neg64 r4
+mov64 r1, r0
+call __udivti3
+jne r1, 0, +0x2e80
+mov64 r1, r6
+add64 r1, -1
+mov64 r3, 0
 mov64 r2, 1
-ldxdw r4, [r10-1736]
-jge r4, r0, +0x490
-jge r7, r1, +0x4a0
-jne r7, r1, +0x4b0
-and64 r2, 1
-jeq r2, 0, +0x4c8
-ja +0x978
-stw [r10-232], 2
-mov64 r1, r10
-add64 r1, -200
-mov64 r6, r10
-add64 r6, -232
-mov64 r2, r6
-call mayhem_program::MayhemError::name#1311
-stdw [r10-32], 0
-stdw [r10-40], 1
-stdw [r10-48], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#35e0, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#5fac]
-stxdw [r10-368], r1
-mov64 r1, r10
-add64 r1, -48
-stxdw [r10-376], r1
-lddw r1, 0xe0000020
-stxdw [r10-360], r1
-mov64 r2, r10
-add64 r2, -376
-mov64 r1, r6
-call orig_fn_2fcb8
-jne r0, 0, +0x4258
-ldxdw r1, [r10-32]
-stxdw [r10-1656], r1
-ldxdw r1, [r10-40]
-stxdw [r10-1664], r1
-ldxdw r1, [r10-48]
-stxdw [r10-1672], r1
-ldxdw r1, [r10-200]
-stxdw [r10-1696], r1
-ldxdw r1, [r10-192]
-stxdw [r10-1688], r1
-ldxdw r1, [r10-184]
-stxdw [r10-1680], r1
-stw [r10-1576], 6002
-ja +0xab0
+jgt r1, r6, +0xd58
 mov64 r2, 0
-jlt r7, r1, +0x340
-mov64 r3, 0
-jeq r7, r1, +0x348
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x978
-ldxdw r5, [r10-1736]
-mov64 r4, r5
-add64 r4, r0
-mov64 r3, 1
-mov64 r2, 1
-jge r4, r5, +0x538
-mov64 r4, r7
-add64 r4, r1
-add64 r4, r2
-jge r4, r7, +0x560
-jne r4, r7, +0x570
-and64 r2, 1
-jeq r2, 0, +0x588
-ja +0x978
-mov64 r2, 0
-mov64 r4, r7
-add64 r4, r1
-add64 r4, r2
-jlt r4, r7, +0x518
-mov64 r3, 0
-jeq r4, r7, +0x520
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x978
-mov64 r3, r0
-mov64 r4, 1
-ldxdw r5, [r10-1736]
-jlt r5, r3, +0x5b0
-mov64 r4, 0
-mov64 r2, r6
-mov64 r5, r7
-sub64 r5, r1
-sub64 r5, r4
-ldxdw r6, [r10-1736]
-sub64 r6, r0
-mov64 r4, 1
-ldxdw r9, [r10-1792]
-jge r9, r6, +0x8c8
-mov64 r6, 1
-ldxdw r9, [r10-1768]
-jge r9, r5, +0x8e8
-ldxdw r9, [r10-1768]
-jne r9, r5, +0x900
-ldxdw r5, [r10-1736]
-add64 r0, r5
-mov64 r5, 1
-jlt r0, r3, +0x648
-mov64 r5, 0
-mov64 r3, 12
-stxdw [r10-1776], r3
-and64 r4, 1
-mov64 r9, r2
-jne r4, 0, +0x978
-add64 r1, r7
-add64 r1, r5
-mov64 r3, 1
-mov64 r2, 1
-ldxdw r4, [r10-1792]
-jgt r4, r0, +0x930
-ldxdw r4, [r10-1768]
-jgt r4, r1, +0x948
-ldxdw r4, [r10-1768]
-jne r4, r1, +0x960
-and64 r2, 1
-jeq r2, 0, +0x978
+jeq r2, 0, +0xd60
+stxdw [r10-2936], r3
+stxdw [r10-2736], r0
+jeq r7, 0, +0xd80
+ldxdw r1, [r9+48]
+call solana_account_info::AccountInfo::lamports#aa02
+stxdw [r10-2952], r0
 ldxdw r1, [r9+32]
 ldxdw r1, [r1+104]
-ldxdw r2, [r9+8]
-ldxdw r3, [r9+104]
-ldxdw r3, [r3]
-ldxdw r4, [r3+24]
-stxdw [r10-1704], r4
-ldxdw r4, [r3+16]
-stxdw [r10-1712], r4
-ldxdw r4, [r3+8]
-stxdw [r10-1720], r4
-ldxdw r3, [r3]
-stxdw [r10-1728], r3
+stxdw [r10-2944], r1
+ja +0xd98
+jne r2, 0, +0xd08
+mov64 r3, r1
+stxdw [r10-2936], r3
+stxdw [r10-2736], r0
+jne r7, 0, +0xd20
+ldxdw r1, [r9+40]
+ldxdw r1, [r1+104]
+stxdw [r10-2952], r1
+ldxdw r1, [r10-2776]
+ldxb r1, [r1+34]
+stxb [r10-2721], r1
+mov64 r1, r10
+add64 r1, -2721
+stxdw [r10-2704], r1
+lddw r1, .rodata[736f6c2d7661756c]
+stxdw [r10-2720], r1
+stdw [r10-2696], 1
+stdw [r10-2712], 9
+ldxdw r1, [r9+88]
+ldxdw r2, [r1]
+mov64 r1, r10
+add64 r1, -1584
+stxdw [r10-2744], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+48]
+mov64 r1, r10
+add64 r1, -1536
+stxdw [r10-2768], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+96]
+ldxdw r2, [r1]
+mov64 r1, r10
+add64 r1, -1488
+stxdw [r10-2776], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+24]
+ldxdw r2, [r1+88]
+mov64 r1, r10
+add64 r1, -1440
+stxdw [r10-2784], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+104]
+mov64 r1, r10
+add64 r1, -1392
+stxdw [r10-2800], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+32]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -1344
+stxdw [r10-2808], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+40]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -1296
+stxdw [r10-2816], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+112]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -1248
+stxdw [r10-2824], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+120]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -1200
+stxdw [r10-2832], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+128]
+mov64 r1, r10
+add64 r1, -1152
+stxdw [r10-2840], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+136]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -1104
+stxdw [r10-2848], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+72]
+mov64 r1, r10
+add64 r1, -1056
+stxdw [r10-2856], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+64]
+mov64 r1, r10
+add64 r1, -1008
+stxdw [r10-2864], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+56]
+mov64 r1, r10
+add64 r1, -960
+stxdw [r10-2872], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+80]
+mov64 r1, r10
+add64 r1, -912
+stxdw [r10-2880], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+144]
+mov64 r1, r10
+add64 r1, -864
+stxdw [r10-2888], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+152]
+mov64 r1, r10
+add64 r1, -816
+stxdw [r10-2896], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r9+160]
+ldxdw r2, [r1+32]
+mov64 r6, r10
+add64 r6, -696
+mov64 r1, r6
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+168]
+mov64 r1, r10
+add64 r1, -648
+stxdw [r10-2904], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+176]
+mov64 r8, r10
+add64 r8, -600
+mov64 r1, r8
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+184]
+mov64 r1, r10
+add64 r1, -56
+stxdw [r10-2912], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+216]
+mov64 r1, r10
+add64 r1, -136
+stxdw [r10-2920], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+224]
+mov64 r1, r10
+add64 r1, -552
+stxdw [r10-2928], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r9, r10
+add64 r9, -2688
+mov64 r1, r9
+ldxdw r2, [r10-2744]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2640
+ldxdw r2, [r10-2768]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2592
+ldxdw r2, [r10-2776]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2544
+ldxdw r2, [r10-2784]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2496
+ldxdw r2, [r10-2800]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2448
+ldxdw r2, [r10-2808]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2400
+ldxdw r2, [r10-2816]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2352
+ldxdw r2, [r10-2824]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2304
+ldxdw r2, [r10-2832]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2256
+ldxdw r2, [r10-2840]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2208
+ldxdw r2, [r10-2848]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2160
+ldxdw r2, [r10-2856]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2112
+ldxdw r2, [r10-2864]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2064
+ldxdw r2, [r10-2872]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -2016
+ldxdw r2, [r10-2880]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1968
+ldxdw r2, [r10-2888]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1920
+ldxdw r2, [r10-2896]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1872
+mov64 r2, r6
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1824
+ldxdw r2, [r10-2904]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1776
+mov64 r2, r8
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1728
+ldxdw r2, [r10-2912]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1680
+ldxdw r2, [r10-2920]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -1632
+ldxdw r2, [r10-2928]
+mov64 r3, 48
+call memcpy
+stw [r10-853], 0
+sth [r10-855], 256
+stw [r10-859], 16843009
+stb [r10-860], 0
+stw [r10-864], 257
+sth [r10-849], 0
+stb [r10-842], 1
+stw [r10-846], 0
+stb [r10-847], 1
+mov64 r1, r10
+add64 r1, -864
+stxdw [r10-720], r1
+mov64 r1, r10
+add64 r1, -1584
+stxdw [r10-736], r1
+stxdw [r10-744], r9
+stdw [r10-728], 0
+mov64 r1, r10
+add64 r1, -768
+mov64 r2, r10
+add64 r2, -744
+call <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter#52e3
+call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
+lddw r1, 0x300000000
+ldxdw r1, [r1]
+mov64 r2, 0
+mov64 r4, 1
+mov64 r3, r1
+add64 r3, -8
+jgt r3, r1, +0x16d8
+mov64 r4, 0
+ldxdw r5, [r10-2736]
+jeq r4, 0, +0x16e8
+lddw r3, 0x300007ff8
+ldxdw r9, [r10-2792]
+jeq r1, 0, +0x1710
+mov64 r3, r2
+lddw r1, 0x300000008
+jge r3, r1, +0x1728
+mov64 r1, 1
+mov64 r2, 8
+call alloc::alloc::handle_alloc_error#decd
+ldxdw r5, [r10-2736]
+jne r4, 0, +0x1680
+mov64 r2, r3
+lddw r3, 0x300007ff8
+ldxdw r9, [r10-2792]
+jne r1, 0, +0x16a0
+lddw r1, 0x300000008
+jlt r3, r1, +0x16c0
+lddw r1, 0x300000000
+stxdw [r1], r3
+lddw r1, 0xad837f01a485e633
+stxdw [r3], r1
+stxdw [r10-808], r3
+stdw [r10-800], 8
+stdw [r10-816], 8
+stxdw [r10-552], r5
+mov64 r6, r10
+add64 r6, -816
+mov64 r2, r10
+add64 r2, -552
 mov64 r3, r10
-add64 r3, -1728
-stxdw [r10-4080], r3
-ldxdw r3, [r10-1752]
-stxdw [r10-4088], r3
-stxdw [r10-4096], r1
-add64 r2, 8
-mov64 r1, r10
-add64 r1, -376
-mov64 r5, r10
-ldxdw r3, [r10-1800]
-mov64 r4, r8
-call orig_fn_2d798
-ldxdw r1, [r10-360]
-ldxdw r0, [r10-368]
-ldxdw r2, [r10-376]
-jne r2, 0, +0xad8
-ldxdw r2, [r9+104]
-ldxdw r4, [r2]
-ldxdw r5, [r4+24]
-mov64 r3, 1
-lddw r6, 0x100000000f0a098
-mov64 r2, 1
-jne r5, r6, +0xae0
-ldxdw r6, [r4+16]
-lddw r7, 0x553beb1adc39c4da
+add64 r3, -544
+mov64 r1, r6
+call <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<&T,core::slice::iter::Iter<T>>>::spec_extend#7ce4
+ldxdw r1, [r10-2936]
+stxdw [r10-552], r1
+mov64 r2, r10
+add64 r2, -552
+mov64 r3, r10
+add64 r3, -544
+mov64 r1, r6
+call <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<&T,core::slice::iter::Iter<T>>>::spec_extend#7ce4
+mov64 r1, r10
+add64 r1, -2720
+stxdw [r10-712], r1
+ldxdw r1, [r10-800]
+stxdw [r10-512], r1
+ldxdw r1, [r10-808]
+stxdw [r10-520], r1
+ldxdw r1, [r10-816]
+stxdw [r10-528], r1
+lddw r1, 0x76c65e82fcde140c
+stxdw [r10-504], r1
+lddw r1, 0x654065bb18082594
+stxdw [r10-496], r1
+lddw r1, 0xb471d556318d29f4
+stxdw [r10-488], r1
+lddw r1, 0x63a8e9180c09f8d4
+stxdw [r10-480], r1
+ldxdw r1, [r10-768]
+stxdw [r10-552], r1
+ldxdw r1, [r10-760]
+stxdw [r10-544], r1
+ldxdw r1, [r10-752]
+stxdw [r10-536], r1
+stdw [r10-704], 2
+mov64 r1, r10
+add64 r1, -552
+mov64 r2, r10
+add64 r2, -2688
+mov64 r4, r10
+add64 r4, -712
+mov64 r3, 23
 mov64 r5, 1
-jne r6, r7, +0xb10
-ldxdw r6, [r4]
-lddw r7, 0x8481abfe57889b06
-mov64 r8, 1
-jne r6, r7, +0xb40
-ldxdw r4, [r4+8]
-lddw r6, 0x35c01846637f68fb
-jeq r4, r6, +0x878
-mov64 r3, 0
-and64 r8, r3
-and64 r8, r5
-and64 r8, r2
-and64 r8, 1
-stxdw [r10-1920], r1
-stxdw [r10-1832], r0
-jeq r8, 0, +0xb70
+call solana_invoke::invoke_signed#812c
+lsh64 r0, 32
+rsh64 r0, 32
+jne r0, 26, +0x1fb0
+jeq r7, 0, +0x1ef0
+ldxdw r1, [r9+64]
+ldxdw r1, [r1]
+ldxdw r2, [r1+24]
+stxdw [r10-576], r2
+ldxdw r2, [r1+16]
+stxdw [r10-584], r2
+ldxdw r2, [r1+8]
+stxdw [r10-592], r2
+ldxdw r1, [r1]
+stxdw [r10-600], r1
+call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
+lddw r1, 0x300000000
+ldxdw r1, [r1]
+mov64 r2, 0
+mov64 r4, 1
+mov64 r3, r1
+add64 r3, -102
+jgt r3, r1, +0x1a38
+mov64 r4, 0
+jeq r4, 0, +0x1a40
+lddw r6, 0x300007f9a
+jeq r1, 0, +0x1a60
+mov64 r6, r2
+lddw r1, 0x300000008
+jge r6, r1, +0x1a78
+mov64 r1, 1
+mov64 r2, 102
+call alloc::alloc::handle_alloc_error#decd
+jne r4, 0, +0x19e8
+mov64 r2, r3
+lddw r6, 0x300007f9a
+jne r1, 0, +0x1a00
+lddw r1, 0x300000008
+jlt r6, r1, +0x1a20
+lddw r8, 0x300000000
+stxdw [r8], r6
+ldxdw r1, [r9+40]
+ldxdw r1, [r1+32]
+ldxdw r1, [r1]
+ldxdw r2, [r9+48]
+ldxdw r2, [r2]
+ldxdw r3, [r2+24]
+stxdw [r10-32], r3
+ldxdw r3, [r2+16]
+stxdw [r10-40], r3
+ldxdw r3, [r2+8]
+stxdw [r10-48], r3
+ldxdw r3, [r2]
+stxdw [r10-56], r3
+ldxdw r3, [r2+24]
+stxdw [r10-528], r3
+ldxdw r3, [r2+16]
+stxdw [r10-536], r3
+ldxdw r3, [r2+8]
+stxdw [r10-544], r3
+ldxdw r2, [r2]
+stxdw [r10-552], r2
+ldxdw r2, [r1]
+ldxdw r3, [r1+8]
+ldxdw r4, [r1+16]
+ldxdw r1, [r1+24]
+stxdw [r6+24], r1
+stxdw [r6+16], r4
+stxdw [r6+8], r3
+stxdw [r6], r2
+sth [r6+32], 256
+ldxdw r1, [r10-56]
+stxdw [r6+34], r1
+ldxdw r1, [r10-48]
+stxdw [r6+42], r1
+ldxdw r1, [r10-40]
+stxdw [r6+50], r1
+ldxdw r1, [r10-32]
+stxdw [r6+58], r1
+sth [r6+66], 256
+ldxdw r1, [r10-552]
+stxdw [r6+68], r1
+ldxdw r1, [r10-544]
+stxdw [r6+76], r1
+ldxdw r1, [r10-536]
+stxdw [r6+84], r1
+ldxdw r1, [r10-528]
+stxdw [r6+92], r1
+sth [r6+100], 1
+call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
+ldxdw r2, [r8]
+lddw r1, 0x300007fff
+jeq r2, 0, +0x1c48
+add64 r2, -1
+mov64 r1, r2
+lddw r2, 0x300000008
+jlt r1, r2, +0x2f00
+lddw r2, 0x300000000
+stxdw [r2], r1
+stb [r1], 9
+stxdw [r10-104], r1
+stxdw [r10-128], r6
+ldxdw r1, [r10-576]
+stxdw [r10-64], r1
+ldxdw r1, [r10-584]
+stxdw [r10-72], r1
+ldxdw r1, [r10-592]
+stxdw [r10-80], r1
+ldxdw r1, [r10-600]
+stxdw [r10-88], r1
+stdw [r10-96], 1
+stdw [r10-112], 1
+stdw [r10-120], 3
+stdw [r10-136], 3
+ldxdw r1, [r9+40]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -696
+stxdw [r10-2744], r1
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+48]
+mov64 r8, r10
+add64 r8, -648
+mov64 r1, r8
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+48]
+mov64 r6, r9
+mov64 r9, r10
+add64 r9, -600
+mov64 r1, r9
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+64]
+mov64 r6, r10
+add64 r6, -56
+mov64 r1, r6
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r10
+add64 r1, -552
+stxdw [r10-2736], r1
+ldxdw r2, [r10-2744]
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -504
+mov64 r2, r8
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -456
+mov64 r2, r9
+mov64 r3, 48
+call memcpy
+mov64 r1, r10
+add64 r1, -408
+mov64 r2, r6
+mov64 r3, 48
+call memcpy
+ldxdw r1, [r10-704]
+stxdw [r10-48], r1
+ldxdw r1, [r10-712]
+stxdw [r10-56], r1
+mov64 r1, r10
+add64 r1, -136
+mov64 r4, r10
+add64 r4, -56
+ldxdw r2, [r10-2736]
+mov64 r3, 4
+mov64 r5, 1
+call solana_invoke::invoke_signed#812c
+mov64 r2, r1
+mov64 r1, r0
+lsh64 r1, 32
+rsh64 r1, 32
+jne r1, 26, +0x1f70
+mov64 r1, r10
+add64 r1, -552
+call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 4]>#7fbb
+ldxdw r9, [r10-2792]
+ldxdw r1, [r9+32]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
+jne r0, 2, +0x2cd8
+ldxdw r1, [r9+112]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
+jne r0, 2, +0x2cd8
+ldxdw r1, [r9+120]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
+jne r0, 2, +0x2cd8
+jeq r7, 0, +0x2058
 ldxdw r1, [r9+48]
 call solana_account_info::AccountInfo::lamports#aa02
-ja +0xb80
-mov64 r4, 0
-mov64 r6, 1
-ldxdw r9, [r10-1768]
-jlt r9, r5, +0x610
-mov64 r6, 0
-ldxdw r9, [r10-1768]
-jeq r9, r5, +0x620
-mov64 r4, r6
-ldxdw r5, [r10-1736]
-add64 r0, r5
-mov64 r5, 1
-jge r0, r3, +0x640
-ja +0x648
-mov64 r2, 0
-ldxdw r4, [r10-1768]
-jle r4, r1, +0x6b0
-mov64 r3, 0
-ldxdw r4, [r10-1768]
-jeq r4, r1, +0x6c0
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x6d0
-ldxdw r7, [r10-1776]
-stxw [r10-232], r7
-mov64 r1, r10
-add64 r1, -200
-mov64 r6, r10
-add64 r6, -232
-mov64 r2, r6
-call mayhem_program::MayhemError::name#1311
-stdw [r10-32], 0
-stdw [r10-40], 1
-stdw [r10-48], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#35e0, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#5fac]
-stxdw [r10-368], r1
-mov64 r1, r10
-add64 r1, -48
-stxdw [r10-376], r1
-lddw r1, 0xe0000020
-stxdw [r10-360], r1
-mov64 r2, r10
-add64 r2, -376
-mov64 r1, r6
-call orig_fn_2fcb8
-jne r0, 0, +0x4258
-ldxdw r1, [r10-32]
-stxdw [r10-1656], r1
-ldxdw r1, [r10-40]
-stxdw [r10-1664], r1
-ldxdw r1, [r10-48]
-stxdw [r10-1672], r1
-ldxdw r1, [r10-200]
-stxdw [r10-1696], r1
-ldxdw r1, [r10-192]
-stxdw [r10-1688], r1
-ldxdw r1, [r10-184]
-stxdw [r10-1680], r1
-or64 r7, 6000
-stxw [r10-1576], r7
-stb [r10-1648], 2
-stdw [r10-1728], 2
-mov64 r1, r10
-add64 r1, -1728
+ja +0x2080
+call mayhem_program::instructions::pump_interface::buy::math#3277
+jeq r0, 2, +0x3f8
+ja +0x620
+mov64 r1, r0
+call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
+mov64 r7, r0
+mov64 r6, r1
+mov64 r1, r10
+add64 r1, -552
+call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 4]>#7fbb
+ja +0x2ce8
+mov64 r1, r10
+add64 r1, -520
+lddw r2, .rodata[0a000000]
+call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#3c69
+mov64 r1, r10
+add64 r1, -496
+mov64 r2, 10
+call <T as alloc::string::SpecToString>::spec_to_string#08c4
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-544], r1
+stb [r10-472], 2
+stw [r10-528], 326
+stdw [r10-536], 62
+stdw [r10-552], 0
+stw [r10-400], 6010
+mov64 r1, r10
+add64 r1, -552
 call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-exit
-mov64 r2, 0
-ldxdw r6, [r4+16]
-lddw r7, 0x553beb1adc39c4da
-mov64 r5, 1
-jeq r6, r7, +0x828
-mov64 r5, 0
-ldxdw r6, [r4]
-lddw r7, 0x8481abfe57889b06
-mov64 r8, 1
-jeq r6, r7, +0x850
-mov64 r8, 0
-ldxdw r4, [r4+8]
-lddw r6, 0x35c01846637f68fb
-jne r4, r6, +0x870
-ja +0x878
+ja +0x2cd8
+ldxdw r1, [r9+40]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
+jne r0, 2, +0x2cd8
 ldxdw r1, [r9+40]
 ldxdw r0, [r1+104]
+ldxdw r3, [r10-2952]
+jlt r0, r3, +0x2f18
 ldxdw r1, [r9+32]
-ldxdw r5, [r1+104]
-ldxdw r1, [r9+152]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-1704], r2
-ldxdw r2, [r1+16]
-stxdw [r10-1712], r2
-ldxdw r2, [r1+8]
-stxdw [r10-1720], r2
-ldxdw r1, [r1]
-stxdw [r10-1728], r1
-ldxdw r2, [r9+48]
-ldxdw r7, [r2+8]
-ldxdw r1, [r7]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xc18
-mov64 r3, 0
-ldxdw r6, [r2]
-stxdw [r7], r1
-jeq r3, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xc60
-mov64 r3, 0
-stxdw [r10-1928], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-stxdw [r10-1936], r6
-ldxdw r1, [r9+88]
-ldxdw r1, [r1]
-ldxdw r3, [r1+8]
-stxdw [r10-1736], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xcc8
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-1976], r6
-ldxb r6, [r2+41]
-stxdw [r10-1968], r6
-ldxb r6, [r2+40]
-stxdw [r10-1960], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1952], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1944], r2
-ldxdw r6, [r1]
-ldxdw r2, [r10-1736]
+ldxdw r1, [r1+104]
+ldxdw r2, [r10-2944]
+jlt r2, r1, +0x2f30
+mov64 r8, r0
+sub64 r8, r3
+ldxdw r2, [r9+8]
+ldxdw r3, [r10-2752]
+jeq r3, 0, +0x2150
+ldxdw r5, [r2+80]
+ldxdw r4, [r2+72]
+mov64 r6, 1
+jlt r4, r8, +0x2100
+mov64 r6, 0
+mov64 r3, r5
+sub64 r3, r6
+mov64 r6, r5
+xor64 r6, r3
+and64 r5, r6
+jslt r5, 0, +0x2cd0
+sub64 r4, r8
+stxdw [r2+72], r4
+mov64 r4, 80
+ja +0x21c0
+ldxdw r5, [r2+96]
+ldxdw r4, [r2+88]
+mov64 r6, 1
+jlt r4, r8, +0x2178
+mov64 r6, 0
+mov64 r3, r5
+sub64 r3, r6
+mov64 r6, r5
+xor64 r6, r3
+and64 r5, r6
+jslt r5, 0, +0x2cd0
+sub64 r4, r8
+stxdw [r2+88], r4
+mov64 r4, 96
+add64 r2, r4
 stxdw [r2], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xd68
-mov64 r3, 0
-stxdw [r10-1984], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-stxdw [r10-3176], r5
-ldxdw r2, [r9+96]
+ldxdw r2, [r9+16]
+ldxdw r3, [r2+40]
+mov64 r6, 1
+jlt r3, r8, +0x21f8
+mov64 r6, 0
+ldxdw r5, [r2+48]
+mov64 r4, r5
+sub64 r4, r6
+mov64 r6, r5
+xor64 r6, r4
+and64 r5, r6
+jslt r5, 0, +0x2cd0
+ldxdw r5, [r10-2944]
+sub64 r5, r1
+sub64 r3, r8
+stxdw [r2+40], r3
+stxdw [r2+48], r4
+ldxdw r1, [r9+16]
+ldxdw r3, [r1+56]
+mov64 r2, r3
+stxdw [r10-2944], r5
+add64 r2, r5
+mov64 r5, 1
+jlt r2, r3, +0x2298
+mov64 r5, 0
+ldxdw r4, [r1+64]
+mov64 r3, r4
+add64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+xor64 r4, -1
+and64 r4, r5
+jsle r4, -1, +0x2cd0
+stxdw [r1+56], r2
+stxdw [r1+64], r3
+ldxdw r1, [r9+48]
+ldxdw r2, [r1]
+ldxdw r3, [r2+24]
+stxdw [r10-576], r3
+ldxdw r3, [r2+16]
+stxdw [r10-584], r3
+ldxdw r3, [r2+8]
+stxdw [r10-592], r3
 ldxdw r2, [r2]
-ldxdw r3, [r2+8]
-stxdw [r10-1744], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xdd0
-mov64 r4, 0
-ldxb r5, [r1+42]
-stxdw [r10-2032], r5
-ldxb r5, [r1+41]
-stxdw [r10-2024], r5
-ldxb r5, [r1+40]
-stxdw [r10-2016], r5
-ldxdw r5, [r1+32]
-stxdw [r10-2008], r5
-ldxdw r1, [r1+24]
-stxdw [r10-2000], r1
-ldxdw r1, [r2]
-stxdw [r10-1992], r1
-ldxdw r1, [r10-1744]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xe78
-mov64 r3, 0
-stxdw [r10-2040], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-stxdw [r10-3184], r0
-stxdw [r10-2048], r6
-ldxdw r1, [r9+112]
-ldxdw r1, [r1+32]
-ldxdw r3, [r1+8]
-stxdw [r10-1768], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xee8
-mov64 r4, 0
-ldxb r5, [r2+42]
-stxdw [r10-2064], r5
-ldxb r6, [r2+41]
-ldxb r5, [r2+40]
-stxdw [r10-2056], r5
-ldxdw r0, [r2+32]
-stxdw [r10-2072], r0
-ldxdw r0, [r2+24]
-ldxdw r5, [r1]
-ldxdw r2, [r10-1768]
-stxdw [r2], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xf78
-mov64 r3, 0
-stxdw [r10-2080], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-stxdw [r10-2136], r6
-ldxdw r2, [r9+120]
-ldxdw r2, [r2+32]
-ldxdw r3, [r2+8]
-stxdw [r10-1776], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xfe0
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-2128], r6
-ldxb r6, [r1+41]
-stxdw [r10-2120], r6
-ldxb r6, [r1+40]
-stxdw [r10-2112], r6
-ldxdw r6, [r1+32]
-stxdw [r10-2104], r6
-ldxdw r1, [r1+24]
-stxdw [r10-2096], r1
-ldxdw r1, [r2]
-stxdw [r10-2088], r1
-ldxdw r1, [r10-1776]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1088
-mov64 r3, 0
-stxdw [r10-2144], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-stxdw [r10-2152], r0
-ldxdw r1, [r9+32]
-ldxdw r1, [r1+32]
-ldxdw r3, [r1+8]
-stxdw [r10-1792], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x10f0
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-2184], r6
-ldxb r6, [r2+41]
-stxdw [r10-2176], r6
-ldxb r6, [r2+40]
-stxdw [r10-2168], r6
-ldxdw r6, [r2+32]
-stxdw [r10-2160], r6
-ldxdw r0, [r2+24]
-ldxdw r2, [r1]
-stxdw [r10-2192], r2
-ldxdw r2, [r10-1792]
-stxdw [r2], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1190
-mov64 r3, 0
-stxdw [r10-2200], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-stxdw [r10-2208], r5
-ldxdw r2, [r9+40]
-ldxdw r2, [r2+32]
-ldxdw r3, [r2+8]
-stxdw [r10-1800], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x11f8
-mov64 r4, 0
-ldxb r6, [r1+42]
-ldxb r5, [r1+41]
-stxdw [r10-2240], r5
-ldxb r5, [r1+40]
-stxdw [r10-2248], r5
-ldxdw r5, [r1+32]
-stxdw [r10-2232], r5
-ldxdw r1, [r1+24]
-stxdw [r10-2224], r1
-ldxdw r1, [r2]
-stxdw [r10-2216], r1
-ldxdw r1, [r10-1800]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1298
-mov64 r3, 0
-stxdw [r10-2256], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-ldxdw r1, [r9+64]
-ldxdw r3, [r1+8]
-stxdw [r10-1808], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x12f0
-mov64 r4, 0
-ldxb r5, [r2+42]
-stxdw [r10-2296], r5
-ldxb r5, [r2+41]
-stxdw [r10-2288], r5
-ldxb r5, [r2+40]
-stxdw [r10-2280], r5
-ldxdw r5, [r2+32]
-stxdw [r10-2272], r5
-ldxdw r5, [r2+24]
-ldxdw r2, [r1]
-stxdw [r10-2264], r2
-ldxdw r2, [r10-1808]
-stxdw [r2], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1390
-mov64 r3, 0
-stxdw [r10-2304], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-stxdw [r10-2352], r6
-ldxdw r2, [r9+72]
-ldxdw r3, [r2+8]
-stxdw [r10-1816], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x13f0
-mov64 r4, 0
-stxdw [r10-2360], r5
-ldxb r6, [r1+42]
-stxdw [r10-2344], r6
-ldxb r6, [r1+41]
-stxdw [r10-2336], r6
-ldxb r6, [r1+40]
-stxdw [r10-2328], r6
-ldxdw r6, [r1+32]
-stxdw [r10-2320], r6
-ldxdw r1, [r1+24]
-stxdw [r10-2312], r1
-ldxdw r6, [r2]
-ldxdw r1, [r10-1816]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1498
-mov64 r3, 0
-stxdw [r10-2368], r6
-stxdw [r10-2376], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-stxdw [r10-2384], r0
-ldxdw r3, [r9+56]
-ldxdw r5, [r3+8]
-ldxdw r1, [r5]
-add64 r1, 1
-mov64 r4, 1
-jeq r1, 0, +0x14f8
-mov64 r4, 0
-ldxb r0, [r2+42]
-stxdw [r10-2432], r0
-ldxb r0, [r2+41]
-stxdw [r10-2424], r0
-ldxb r0, [r2+40]
-stxdw [r10-2416], r0
-ldxdw r0, [r2+32]
-stxdw [r10-2408], r0
-ldxdw r2, [r2+24]
-stxdw [r10-2400], r2
-ldxdw r2, [r3]
-stxdw [r10-2392], r2
-stxdw [r5], r1
-jeq r4, 1, +0x3a58
-ldxdw r4, [r3+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x1598
-mov64 r2, 0
-stxdw [r10-2440], r4
-stxdw [r4], r1
-jeq r2, 1, +0x3a58
-stxdw [r10-1848], r8
-ldxdw r1, [r9+104]
-ldxdw r2, [r1+8]
-stxdw [r10-1824], r2
-ldxdw r2, [r2]
-add64 r2, 1
-mov64 r4, 1
-jeq r2, 0, +0x15f8
-mov64 r4, 0
-ldxb r0, [r3+42]
-stxdw [r10-2472], r0
-ldxb r0, [r3+41]
-stxdw [r10-2464], r0
-ldxb r0, [r3+40]
-stxdw [r10-2456], r0
-ldxdw r0, [r3+32]
-stxdw [r10-2448], r0
-ldxdw r0, [r3+24]
-ldxdw r8, [r1]
-ldxdw r3, [r10-1824]
-stxdw [r3], r2
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1690
-mov64 r3, 0
-stxdw [r10-2480], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
+stxdw [r10-600], r2
 ldxdw r2, [r9+24]
-ldxdw r2, [r2+88]
-ldxdw r3, [r2+8]
-stxdw [r10-1840], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x16f0
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-2488], r6
-ldxb r6, [r1+41]
-stxdw [r10-2512], r6
-ldxb r6, [r1+40]
-stxdw [r10-2520], r6
-ldxdw r6, [r1+32]
-stxdw [r10-2504], r6
-ldxdw r1, [r1+24]
-stxdw [r10-2496], r1
-ldxdw r6, [r2]
-ldxdw r1, [r10-1840]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1790
-mov64 r3, 0
-stxdw [r10-2528], r0
-stxdw [r10-2536], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-mov64 r0, r7
-ldxdw r1, [r9+80]
-ldxdw r3, [r1+8]
-stxdw [r10-1856], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x17f8
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2568], r7
-ldxb r7, [r2+41]
-stxdw [r10-2560], r7
-ldxb r7, [r2+40]
-stxdw [r10-2552], r7
-ldxdw r7, [r2+32]
-ldxdw r2, [r2+24]
-stxdw [r10-2576], r2
-ldxdw r2, [r1]
-stxdw [r10-2544], r2
-ldxdw r2, [r10-1856]
-stxdw [r2], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1898
-mov64 r3, 0
-stxdw [r10-2592], r8
-stxdw [r10-2584], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-stxdw [r10-2648], r7
-ldxdw r2, [r9+144]
-ldxdw r3, [r2+8]
-stxdw [r10-1864], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-ldxdw r7, [r10-1848]
-jeq r3, 0, +0x1908
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2640], r7
-ldxb r7, [r1+41]
-stxdw [r10-2632], r7
-ldxb r7, [r1+40]
-stxdw [r10-2624], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2616], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2608], r1
-ldxdw r1, [r2]
-stxdw [r10-2600], r1
-ldxdw r1, [r10-1864]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-mov64 r8, r0
-jeq r1, 0, +0x19b8
-mov64 r3, 0
-stxdw [r10-2656], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-ldxdw r3, [r9+152]
-ldxdw r1, [r3+8]
-stxdw [r10-1872], r1
-ldxdw r1, [r1]
-add64 r1, 1
-mov64 r4, 1
-jeq r1, 0, +0x1a10
-mov64 r4, 0
-ldxb r0, [r2+42]
-stxdw [r10-2696], r0
-ldxb r0, [r2+41]
-stxdw [r10-2688], r0
-ldxb r0, [r2+40]
-stxdw [r10-2680], r0
-ldxdw r0, [r2+32]
-stxdw [r10-2672], r0
-ldxdw r2, [r2+24]
-stxdw [r10-2664], r2
-ldxdw r0, [r3]
-ldxdw r2, [r10-1872]
-stxdw [r2], r1
-jeq r4, 1, +0x3a58
-ldxdw r4, [r3+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x1ab0
-mov64 r2, 0
-stxdw [r10-2704], r4
-stxdw [r4], r1
-jeq r2, 1, +0x3a58
-stxdw [r10-2752], r6
-ldxdw r1, [r9+128]
-ldxdw r2, [r1+8]
-stxdw [r10-1880], r2
-ldxdw r2, [r2]
-add64 r2, 1
-mov64 r4, 1
-jeq r2, 0, +0x1b10
-mov64 r4, 0
-ldxb r6, [r3+42]
-stxdw [r10-2744], r6
-ldxb r6, [r3+41]
-stxdw [r10-2736], r6
-ldxb r6, [r3+40]
-stxdw [r10-2728], r6
-ldxdw r6, [r3+32]
-stxdw [r10-2720], r6
-ldxdw r3, [r3+24]
-stxdw [r10-2712], r3
-ldxdw r3, [r1]
-stxdw [r10-2760], r3
-ldxdw r3, [r10-1880]
-stxdw [r3], r2
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-mov64 r6, r5
-jeq r2, 0, +0x1bc0
-mov64 r3, 0
-stxdw [r10-2768], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-ldxdw r2, [r9+136]
-ldxdw r2, [r2+32]
-ldxdw r5, [r2+8]
-ldxdw r3, [r5]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1c18
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2808], r7
-ldxb r7, [r1+41]
-stxdw [r10-2800], r7
-ldxb r7, [r1+40]
-stxdw [r10-2792], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2784], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2776], r1
-ldxdw r7, [r2]
-stxdw [r5], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1cb0
-mov64 r3, 0
-stxdw [r10-2816], r5
-stxdw [r10-2824], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-stxdw [r10-2832], r8
-ldxdw r1, [r9+160]
-ldxdw r1, [r1+32]
-ldxdw r5, [r1+8]
-ldxdw r3, [r5]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1d18
-mov64 r4, 0
-ldxb r8, [r2+42]
-stxdw [r10-2872], r8
-ldxb r8, [r2+41]
-stxdw [r10-2864], r8
-ldxb r8, [r2+40]
-stxdw [r10-2856], r8
-ldxdw r8, [r2+32]
-stxdw [r10-2848], r8
-ldxdw r8, [r2+24]
-ldxdw r2, [r1]
-stxdw [r10-2840], r2
-stxdw [r5], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1db0
-mov64 r3, 0
-stxdw [r10-2880], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-stxdw [r10-2936], r7
-ldxdw r2, [r9+168]
-ldxdw r3, [r2+8]
-stxdw [r10-1888], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1e10
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2928], r7
-ldxb r7, [r1+41]
-stxdw [r10-2920], r7
-ldxb r7, [r1+40]
-stxdw [r10-2912], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2904], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2896], r1
-ldxdw r1, [r2]
-stxdw [r10-2888], r1
-ldxdw r1, [r10-1888]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1eb8
-mov64 r3, 0
-stxdw [r10-2944], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-ldxdw r1, [r9+176]
-ldxdw r3, [r1+8]
-stxdw [r10-1896], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1f10
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2984], r7
-ldxb r7, [r2+41]
-stxdw [r10-2976], r7
-ldxb r7, [r2+40]
-stxdw [r10-2968], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2960], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2952], r2
-ldxdw r7, [r1]
-ldxdw r2, [r10-1896]
-stxdw [r2], r3
-jeq r4, 1, +0x3a58
-stxdw [r10-1912], r7
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1fb8
-mov64 r3, 0
-mov64 r7, r8
-stxdw [r10-2992], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3a58
-mov64 r8, r6
-ldxdw r2, [r9+184]
-ldxdw r3, [r2+8]
-stxdw [r10-1904], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x2020
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-3040], r6
-ldxb r6, [r1+41]
-stxdw [r10-3032], r6
-ldxb r6, [r1+40]
-stxdw [r10-3024], r6
-ldxdw r6, [r1+32]
-stxdw [r10-3016], r6
-ldxdw r1, [r1+24]
-stxdw [r10-3008], r1
-ldxdw r1, [r2]
-stxdw [r10-3000], r1
-ldxdw r1, [r10-1904]
-stxdw [r1], r3
-jeq r4, 1, +0x3a58
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x20c8
-mov64 r3, 0
-stxdw [r10-3056], r5
-stxdw [r10-3048], r0
-stxdw [r10-3064], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3a58
-ldxb r1, [r2+42]
-stxdw [r10-3096], r1
-ldxb r1, [r2+41]
-stxdw [r10-3088], r1
-ldxb r1, [r2+40]
-stxdw [r10-3080], r1
-ldxdw r1, [r2+32]
-stxdw [r10-3072], r1
-ldxdw r6, [r2+24]
-ldxdw r1, [r10-1760]
-ldxb r1, [r1+34]
-mov64 r2, r10
-add64 r2, -377
-stxdw [r10-360], r2
-lddw r2, .rodata[736f6c2d7661756c]
-stxdw [r10-376], r2
-stxb [r10-377], r1
-mov64 r1, r10
-add64 r1, -376
-stxdw [r10-200], r1
-ldxw r1, [r10-232]
-stxw [r10-484], r1
-ldxb r1, [r10-228]
-stxb [r10-480], r1
-stdw [r10-352], 1
-stdw [r10-368], 9
-stdw [r10-192], 2
-ldxw r1, [r10-382]
-stxw [r10-492], r1
-ldxb r1, [r10-378]
-stxb [r10-488], r1
-ldxw r1, [r10-387]
-stxw [r10-500], r1
-ldxb r1, [r10-383]
-stxb [r10-496], r1
-ldxb r1, [r10-388]
-stxb [r10-504], r1
-ldxw r1, [r10-392]
-stxw [r10-508], r1
-ldxb r1, [r10-393]
-stxb [r10-512], r1
-ldxw r1, [r10-397]
-stxw [r10-516], r1
-ldxb r1, [r10-398]
-stxb [r10-520], r1
-ldxw r1, [r10-402]
-stxw [r10-524], r1
-ldxb r1, [r10-403]
-stxb [r10-528], r1
-ldxw r1, [r10-407]
-stxw [r10-532], r1
-ldxb r1, [r10-408]
-stxb [r10-536], r1
-ldxw r1, [r10-412]
-stxw [r10-540], r1
-ldxb r1, [r10-413]
-stxb [r10-544], r1
-ldxw r1, [r10-417]
-stxw [r10-548], r1
-ldxb r1, [r10-418]
-stxb [r10-552], r1
-ldxw r1, [r10-422]
-stxw [r10-556], r1
-ldxb r1, [r10-423]
-stxb [r10-560], r1
-ldxw r1, [r10-427]
-stxw [r10-564], r1
-ldxb r1, [r10-428]
-stxb [r10-568], r1
-ldxw r1, [r10-432]
-stxw [r10-572], r1
-ldxb r1, [r10-433]
-stxb [r10-576], r1
-ldxw r1, [r10-437]
-stxw [r10-580], r1
-ldxb r1, [r10-438]
-stxb [r10-584], r1
-ldxw r1, [r10-442]
-stxw [r10-588], r1
-ldxb r1, [r10-443]
-stxb [r10-592], r1
-ldxw r1, [r10-447]
-stxw [r10-596], r1
-ldxb r1, [r10-448]
-stxb [r10-600], r1
-ldxw r1, [r10-452]
-stxw [r10-604], r1
-ldxb r1, [r10-453]
-stxb [r10-608], r1
-ldxw r1, [r10-457]
-stxw [r10-612], r1
-ldxb r1, [r10-458]
-stxb [r10-616], r1
-ldxw r1, [r10-462]
-stxw [r10-620], r1
-ldxb r1, [r10-463]
-stxb [r10-624], r1
-ldxw r1, [r10-467]
-stxw [r10-628], r1
-ldxb r1, [r10-468]
-stxb [r10-632], r1
-ldxw r1, [r10-472]
-stxw [r10-636], r1
-ldxb r1, [r10-473]
-stxb [r10-640], r1
-ldxw r1, [r10-477]
-stxw [r10-644], r1
-ldxdw r1, [r10-1704]
-stxdw [r10-24], r1
-ldxdw r1, [r10-1712]
-stxdw [r10-32], r1
-ldxdw r1, [r10-1720]
-stxdw [r10-40], r1
-ldxdw r1, [r10-1728]
-stxdw [r10-48], r1
-call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
-lddw r1, 0x300000000
-ldxdw r3, [r1]
-mov64 r2, 0
-mov64 r4, 1
-mov64 r1, r3
-add64 r1, -96
-jle r1, r3, +0x35a8
-ldxdw r5, [r10-1832]
-jeq r4, 0, +0x35c0
-lddw r1, 0x300007fa0
-jne r3, 0, +0x35e0
-lddw r2, 0x300000008
-jlt r1, r2, +0x3608
-lddw r2, 0x300000000
-stxdw [r2], r1
-ldxdw r0, [r9+216]
-ldxdw r3, [r0+8]
-ldxdw r2, [r3]
-add64 r2, 1
-mov64 r4, 1
-jeq r2, 0, +0x2568
-mov64 r4, 0
-ldxdw r5, [r0]
-stxdw [r10-1760], r5
-stxdw [r3], r2
-and64 r4, 1
-jne r4, 0, +0x3a58
-ldxdw r5, [r0+16]
-ldxdw r2, [r5]
-add64 r2, 1
-mov64 r4, 1
-jeq r2, 0, +0x25c0
-mov64 r4, 0
-stxdw [r10-3120], r6
-stxdw [r10-3112], r7
-stxdw [r10-3104], r8
-stxdw [r5], r2
-jeq r4, 1, +0x3a58
-ldxdw r9, [r9+224]
-ldxdw r8, [r9+8]
-ldxdw r4, [r8]
-add64 r4, 1
-mov64 r6, 1
-ldxdw r2, [r10-1912]
-jeq r4, 0, +0x2628
-mov64 r6, 0
-ldxb r2, [r0+42]
-stxdw [r10-3144], r2
-ldxb r2, [r0+41]
-stxdw [r10-3136], r2
-ldxb r2, [r0+40]
-stxdw [r10-3128], r2
-ldxdw r7, [r0+32]
-ldxdw r2, [r0+24]
-ldxdw r0, [r9]
-stxdw [r10-3152], r0
-stxdw [r8], r4
-jeq r6, 1, +0x3a58
-stxdw [r10-3160], r2
-ldxdw r4, [r9+16]
-ldxdw r6, [r4]
-add64 r6, 1
-mov64 r0, 1
-jeq r6, 0, +0x26c0
-mov64 r0, 0
-mov64 r2, r7
-stxdw [r10-3168], r3
-stxdw [r4], r6
-jeq r0, 1, +0x3a58
-ldxdw r0, [r9+24]
-ldxdw r6, [r9+32]
-ldxb r7, [r9+40]
-ldxb r3, [r9+41]
-ldxb r9, [r9+42]
-stxb [r1+90], r9
-stxb [r1+89], r3
-stxb [r1+88], r7
-stxdw [r1+80], r6
-stxdw [r1+72], r0
-stxdw [r1+64], r4
-stxdw [r1+56], r8
-ldxdw r3, [r10-3152]
-stxdw [r1+48], r3
-ldxdw r3, [r10-3144]
-stxb [r1+42], r3
-ldxdw r3, [r10-3136]
-stxb [r1+41], r3
-ldxdw r3, [r10-3128]
-stxb [r1+40], r3
-stxdw [r1+32], r2
-ldxdw r3, [r10-3160]
-stxdw [r1+24], r3
-stxdw [r1+16], r5
-ldxdw r2, [r10-3168]
-stxdw [r1+8], r2
-ldxdw r2, [r10-1760]
-stxdw [r1], r2
-ldxdw r2, [r10-2032]
-stxb [r10-1662], r2
-ldxdw r2, [r10-2024]
-stxb [r10-1663], r2
-ldxdw r2, [r10-2016]
-stxb [r10-1664], r2
-ldxdw r2, [r10-2008]
-stxdw [r10-1672], r2
-ldxdw r2, [r10-2000]
-stxdw [r10-1680], r2
-ldxdw r2, [r10-1984]
-stxdw [r10-1688], r2
-ldxdw r2, [r10-1736]
-stxdw [r10-1696], r2
-ldxdw r2, [r10-2048]
-stxdw [r10-1704], r2
-stxdw [r10-1720], r1
-ldxdw r1, [r10-1976]
-stxb [r10-1614], r1
-ldxdw r1, [r10-1968]
-stxb [r10-1615], r1
-ldxdw r1, [r10-1960]
-stxb [r10-1616], r1
-ldxdw r1, [r10-1952]
-stxdw [r10-1624], r1
-ldxdw r1, [r10-1944]
-stxdw [r10-1632], r1
-ldxdw r1, [r10-1928]
-stxdw [r10-1640], r1
-ldxdw r1, [r10-2832]
-stxdw [r10-1648], r1
-ldxdw r1, [r10-1936]
-stxdw [r10-1656], r1
-ldxw r1, [r10-484]
-stxw [r10-1661], r1
-ldxb r1, [r10-480]
-stxb [r10-1657], r1
-stdw [r10-1712], 2
-stdw [r10-1728], 2
-ldxdw r1, [r10-1992]
-stxdw [r10-1608], r1
-ldxdw r1, [r10-1744]
-stxdw [r10-1600], r1
-ldxdw r1, [r10-2040]
-stxdw [r10-1592], r1
-ldxdw r1, [r10-2152]
-stxdw [r10-1584], r1
-ldxdw r1, [r10-2072]
-stxdw [r10-1576], r1
-ldxdw r1, [r10-2064]
-stxb [r10-1566], r1
-ldxdw r1, [r10-2136]
-stxb [r10-1567], r1
-ldxdw r1, [r10-2056]
-stxb [r10-1568], r1
-ldxdw r1, [r10-2752]
-stxdw [r10-1560], r1
-ldxdw r1, [r10-1840]
-stxdw [r10-1552], r1
-ldxdw r1, [r10-2536]
-stxdw [r10-1544], r1
-ldxdw r1, [r10-2576]
-stxdw [r10-1536], r1
-ldxdw r1, [r10-2648]
-stxdw [r10-1528], r1
-ldxdw r1, [r10-2568]
-stxb [r10-1518], r1
-ldxdw r1, [r10-2560]
-stxb [r10-1519], r1
-ldxdw r1, [r10-2552]
-stxb [r10-1520], r1
-ldxb r1, [r10-488]
-stxb [r10-1609], r1
-ldxw r1, [r10-492]
-stxw [r10-1613], r1
-ldxw r1, [r10-500]
-stxw [r10-1565], r1
-ldxb r1, [r10-496]
-stxb [r10-1561], r1
-ldxdw r1, [r10-2520]
-stxb [r10-1472], r1
-ldxdw r1, [r10-2512]
-stxb [r10-1471], r1
-ldxdw r1, [r10-2488]
-stxb [r10-1470], r1
-ldxdw r1, [r10-2504]
-stxdw [r10-1480], r1
-ldxdw r1, [r10-2496]
-stxdw [r10-1488], r1
-ldxdw r1, [r10-2480]
-stxdw [r10-1496], r1
-ldxdw r1, [r10-1824]
-stxdw [r10-1504], r1
-ldxdw r1, [r10-2592]
-stxdw [r10-1512], r1
-ldxdw r1, [r10-2248]
-stxb [r10-1424], r1
-ldxdw r1, [r10-2240]
-stxb [r10-1423], r1
-ldxdw r1, [r10-2352]
-stxb [r10-1422], r1
-ldxdw r1, [r10-2232]
-stxdw [r10-1432], r1
-ldxdw r1, [r10-2224]
-stxdw [r10-1440], r1
-ldxdw r1, [r10-2200]
-stxdw [r10-1448], r1
-ldxdw r1, [r10-1792]
-stxdw [r10-1456], r1
-ldxdw r1, [r10-2192]
-stxdw [r10-1464], r1
-ldxb r1, [r10-504]
-stxb [r10-1513], r1
-ldxw r1, [r10-508]
-stxw [r10-1517], r1
-ldxw r1, [r10-516]
-stxw [r10-1469], r1
-ldxb r1, [r10-512]
-stxb [r10-1465], r1
-ldxb r1, [r10-520]
-stxb [r10-1417], r1
-ldxw r1, [r10-524]
-stxw [r10-1421], r1
-ldxdw r1, [r10-2296]
-stxb [r10-1374], r1
-ldxdw r1, [r10-2288]
-stxb [r10-1375], r1
-ldxdw r1, [r10-2280]
-stxb [r10-1376], r1
-ldxdw r1, [r10-2272]
-stxdw [r10-1384], r1
-ldxdw r1, [r10-2360]
-stxdw [r10-1392], r1
-ldxdw r1, [r10-2256]
-stxdw [r10-1400], r1
-ldxdw r1, [r10-1800]
-stxdw [r10-1408], r1
-ldxdw r1, [r10-2216]
-stxdw [r10-1416], r1
-ldxb r1, [r10-528]
-stxb [r10-1369], r1
-ldxw r1, [r10-532]
-stxw [r10-1373], r1
-ldxdw r1, [r10-2128]
-stxb [r10-1326], r1
-ldxdw r1, [r10-2120]
-stxb [r10-1327], r1
-ldxdw r1, [r10-2112]
-stxb [r10-1328], r1
-ldxdw r1, [r10-2104]
-stxdw [r10-1336], r1
-ldxdw r1, [r10-2096]
-stxdw [r10-1344], r1
-ldxdw r1, [r10-2080]
-stxdw [r10-1352], r1
-ldxdw r1, [r10-1768]
-stxdw [r10-1360], r1
-ldxdw r1, [r10-2208]
-stxdw [r10-1368], r1
-ldxb r1, [r10-536]
-stxb [r10-1321], r1
-ldxw r1, [r10-540]
-stxw [r10-1325], r1
-ldxdw r1, [r10-2184]
-stxb [r10-1278], r1
-ldxdw r1, [r10-2176]
-stxb [r10-1279], r1
-ldxdw r1, [r10-2168]
-stxb [r10-1280], r1
-ldxdw r1, [r10-2160]
-stxdw [r10-1288], r1
-ldxdw r1, [r10-2384]
-stxdw [r10-1296], r1
-ldxdw r1, [r10-2144]
-stxdw [r10-1304], r1
-ldxdw r1, [r10-1776]
-stxdw [r10-1312], r1
-ldxdw r1, [r10-2088]
-stxdw [r10-1320], r1
-ldxb r1, [r10-544]
-stxb [r10-1273], r1
-ldxw r1, [r10-548]
-stxw [r10-1277], r1
-ldxdw r1, [r10-2808]
-stxb [r10-1230], r1
-ldxdw r1, [r10-2800]
-stxb [r10-1231], r1
-ldxdw r1, [r10-2792]
-stxb [r10-1232], r1
-ldxdw r1, [r10-2784]
-stxdw [r10-1240], r1
-ldxdw r1, [r10-2776]
-stxdw [r10-1248], r1
-ldxdw r1, [r10-2768]
-stxdw [r10-1256], r1
-ldxdw r1, [r10-1880]
-stxdw [r10-1264], r1
-ldxdw r1, [r10-2760]
-stxdw [r10-1272], r1
-ldxb r1, [r10-552]
-stxb [r10-1225], r1
-ldxw r1, [r10-556]
-stxw [r10-1229], r1
-ldxdw r1, [r10-2872]
-stxb [r10-1182], r1
-ldxdw r1, [r10-2864]
-stxb [r10-1183], r1
-ldxdw r1, [r10-2856]
-stxb [r10-1184], r1
-ldxdw r1, [r10-2848]
-stxdw [r10-1192], r1
-ldxdw r1, [r10-3112]
-stxdw [r10-1200], r1
-ldxdw r1, [r10-2824]
-stxdw [r10-1208], r1
-ldxdw r1, [r10-2816]
-stxdw [r10-1216], r1
-ldxdw r1, [r10-2936]
-stxdw [r10-1224], r1
-ldxb r1, [r10-560]
-stxb [r10-1177], r1
-ldxw r1, [r10-564]
-stxw [r10-1181], r1
-ldxdw r1, [r10-2432]
-stxb [r10-1134], r1
-ldxdw r1, [r10-2424]
-stxb [r10-1135], r1
-ldxdw r1, [r10-2416]
-stxb [r10-1136], r1
-ldxdw r1, [r10-2408]
-stxdw [r10-1144], r1
-ldxdw r1, [r10-2400]
-stxdw [r10-1152], r1
-ldxdw r1, [r10-2376]
-stxdw [r10-1160], r1
-ldxdw r1, [r10-1816]
-stxdw [r10-1168], r1
-ldxdw r1, [r10-2368]
-stxdw [r10-1176], r1
-ldxb r1, [r10-568]
-stxb [r10-1129], r1
-ldxw r1, [r10-572]
-stxw [r10-1133], r1
-ldxdw r1, [r10-2344]
-stxb [r10-1086], r1
-ldxdw r1, [r10-2336]
-stxb [r10-1087], r1
-ldxdw r1, [r10-2328]
-stxb [r10-1088], r1
-ldxdw r1, [r10-2320]
-stxdw [r10-1096], r1
-ldxdw r1, [r10-2312]
-stxdw [r10-1104], r1
-ldxdw r1, [r10-2304]
-stxdw [r10-1112], r1
-ldxdw r1, [r10-1808]
-stxdw [r10-1120], r1
-ldxdw r1, [r10-2264]
-stxdw [r10-1128], r1
-ldxb r1, [r10-576]
-stxb [r10-1081], r1
-ldxw r1, [r10-580]
-stxw [r10-1085], r1
-ldxdw r1, [r10-2472]
-stxb [r10-1038], r1
-ldxdw r1, [r10-2464]
-stxb [r10-1039], r1
-ldxdw r1, [r10-2456]
-stxb [r10-1040], r1
-ldxdw r1, [r10-2448]
-stxdw [r10-1048], r1
-ldxdw r1, [r10-2528]
-stxdw [r10-1056], r1
-ldxdw r1, [r10-2440]
-stxdw [r10-1064], r1
-ldxdw r1, [r10-3104]
-stxdw [r10-1072], r1
-ldxdw r1, [r10-2392]
-stxdw [r10-1080], r1
-ldxb r1, [r10-584]
-stxb [r10-1033], r1
-ldxw r1, [r10-588]
-stxw [r10-1037], r1
-ldxdw r1, [r10-2640]
-stxb [r10-990], r1
-ldxdw r1, [r10-2632]
-stxb [r10-991], r1
-ldxdw r1, [r10-2624]
-stxb [r10-992], r1
-ldxdw r1, [r10-2616]
-stxdw [r10-1000], r1
-ldxdw r1, [r10-2608]
-stxdw [r10-1008], r1
-ldxdw r1, [r10-2584]
-stxdw [r10-1016], r1
-ldxdw r1, [r10-1856]
-stxdw [r10-1024], r1
-ldxdw r1, [r10-2544]
-stxdw [r10-1032], r1
-ldxb r1, [r10-592]
-stxb [r10-985], r1
-ldxw r1, [r10-596]
-stxw [r10-989], r1
-ldxdw r1, [r10-2696]
-stxb [r10-942], r1
-ldxdw r1, [r10-2688]
-stxb [r10-943], r1
-ldxdw r1, [r10-2680]
-stxb [r10-944], r1
-ldxdw r1, [r10-2672]
-stxdw [r10-952], r1
-ldxdw r1, [r10-2664]
-stxdw [r10-960], r1
-ldxdw r1, [r10-2656]
-stxdw [r10-968], r1
-ldxdw r1, [r10-1864]
-stxdw [r10-976], r1
-ldxdw r1, [r10-2600]
-stxdw [r10-984], r1
-ldxb r1, [r10-600]
-stxb [r10-937], r1
-ldxw r1, [r10-604]
-stxw [r10-941], r1
-ldxdw r1, [r10-2744]
-stxb [r10-894], r1
-ldxdw r1, [r10-2736]
-stxb [r10-895], r1
-ldxdw r1, [r10-2728]
-stxb [r10-896], r1
-ldxdw r1, [r10-2720]
-stxdw [r10-904], r1
-ldxdw r1, [r10-2712]
-stxdw [r10-912], r1
-ldxdw r1, [r10-2704]
-stxdw [r10-920], r1
-ldxdw r1, [r10-1872]
-stxdw [r10-928], r1
-ldxdw r1, [r10-3048]
-stxdw [r10-936], r1
-ldxb r1, [r10-608]
-stxb [r10-889], r1
-ldxw r1, [r10-612]
-stxw [r10-893], r1
-ldxdw r1, [r10-2928]
-stxb [r10-846], r1
-ldxdw r1, [r10-2920]
-stxb [r10-847], r1
-ldxdw r1, [r10-2912]
-stxb [r10-848], r1
-ldxdw r1, [r10-2904]
-stxdw [r10-856], r1
-ldxdw r1, [r10-2896]
-stxdw [r10-864], r1
-ldxdw r1, [r10-2880]
-stxdw [r10-872], r1
-ldxdw r1, [r10-3056]
-stxdw [r10-880], r1
-ldxdw r1, [r10-2840]
-stxdw [r10-888], r1
-ldxb r1, [r10-616]
-stxb [r10-841], r1
-ldxw r1, [r10-620]
-stxw [r10-845], r1
-ldxdw r1, [r10-2984]
-stxb [r10-798], r1
-ldxdw r1, [r10-2976]
-stxb [r10-799], r1
-ldxdw r1, [r10-2968]
-stxb [r10-800], r1
-ldxdw r1, [r10-2960]
-stxdw [r10-808], r1
-ldxdw r1, [r10-2952]
-stxdw [r10-816], r1
-ldxdw r1, [r10-2944]
-stxdw [r10-824], r1
-ldxdw r1, [r10-1888]
-stxdw [r10-832], r1
-ldxdw r1, [r10-2888]
-stxdw [r10-840], r1
-ldxb r1, [r10-624]
-stxb [r10-793], r1
-ldxw r1, [r10-628]
-stxw [r10-797], r1
-ldxdw r1, [r10-3040]
-stxb [r10-750], r1
-ldxdw r1, [r10-3032]
-stxb [r10-751], r1
-ldxdw r1, [r10-3024]
-stxb [r10-752], r1
-ldxdw r1, [r10-3016]
-stxdw [r10-760], r1
-ldxdw r1, [r10-3008]
-stxdw [r10-768], r1
-ldxdw r1, [r10-2992]
-stxdw [r10-776], r1
-ldxdw r1, [r10-1896]
-stxdw [r10-784], r1
-ldxdw r1, [r10-1912]
-stxdw [r10-792], r1
-ldxb r1, [r10-632]
-stxb [r10-745], r1
-ldxw r1, [r10-636]
-stxw [r10-749], r1
-ldxdw r1, [r10-3096]
-stxb [r10-702], r1
-ldxdw r1, [r10-3088]
-stxb [r10-703], r1
-ldxdw r1, [r10-3080]
-stxb [r10-704], r1
-ldxdw r1, [r10-3072]
-stxdw [r10-712], r1
-ldxdw r1, [r10-3120]
-stxdw [r10-720], r1
-ldxdw r1, [r10-3064]
-stxdw [r10-728], r1
-ldxdw r1, [r10-1904]
-stxdw [r10-736], r1
-ldxdw r1, [r10-3000]
-stxdw [r10-744], r1
-ldxb r1, [r10-640]
-stxb [r10-697], r1
-ldxw r1, [r10-644]
-stxw [r10-701], r1
-mov64 r1, r10
-add64 r1, -200
-stxdw [r10-696], r1
-stdw [r10-688], 1
-ldxdw r1, [r10-24]
-stxdw [r10-656], r1
-ldxdw r1, [r10-32]
-stxdw [r10-664], r1
-ldxdw r1, [r10-40]
-stxdw [r10-672], r1
-ldxdw r1, [r10-48]
-stxdw [r10-680], r1
-mov64 r1, r10
-add64 r1, -1728
-ldxdw r2, [r10-1832]
-ldxdw r3, [r10-1920]
-call orig_fn_76e18
-mov64 r2, r1
-jne r0, 2, +0x3620
-ldxdw r7, [r10-1784]
-ldxdw r8, [r10-1848]
-ja +0x3648
-mov64 r4, 0
-ldxdw r5, [r10-1832]
-jne r4, 0, +0x24e8
-mov64 r2, r1
-lddw r1, 0x300007fa0
-jeq r3, 0, +0x2500
-and64 r2, -8
-mov64 r1, r2
-lddw r2, 0x300000008
-jge r1, r2, +0x2518
-mov64 r1, 8
-mov64 r2, 96
-call alloc::alloc::handle_alloc_error#decd
-mov64 r1, r0
-call orig_fn_1b8e8
-ldxdw r7, [r10-1784]
-ldxdw r8, [r10-1848]
-jne r0, 2, +0xad8
-jeq r8, 0, +0x3800
-ldxdw r1, [r7+64]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-1520], r2
-ldxdw r2, [r1+16]
-stxdw [r10-1528], r2
-ldxdw r2, [r1+8]
-stxdw [r10-1536], r2
-ldxdw r1, [r1]
-stxdw [r10-1544], r1
-ldxdw r1, [r7+40]
-ldxdw r2, [r1+32]
-mov64 r6, r10
-add64 r6, -376
-mov64 r1, r6
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-ldxdw r2, [r7+48]
-mov64 r1, r10
-add64 r1, -328
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-ldxdw r2, [r7+48]
-mov64 r1, r10
-add64 r1, -280
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-mov64 r1, r10
-add64 r1, -48
-stxdw [r10-200], r1
-mov64 r1, r10
-add64 r1, -232
-stxdw [r10-32], r1
-lddw r1, .rodata[736f6c2d7661756c]
-stxdw [r10-48], r1
-ldxb r1, [r10-377]
-stxb [r10-232], r1
-stdw [r10-192], 2
-stdw [r10-24], 1
-stdw [r10-40], 9
-mov64 r1, r10
-add64 r1, -1704
-mov64 r2, r6
-mov64 r3, 144
-call memcpy
-mov64 r1, r10
-add64 r1, -200
-stxdw [r10-1560], r1
-stdw [r10-1552], 1
-stdw [r10-1712], 0
-stdw [r10-1720], 8
-stdw [r10-1728], 0
-mov64 r1, r10
-add64 r1, -1728
-call orig_fn_91128
-jne r0, 2, +0xad8
-ldxdw r1, [r7+32]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
-jne r0, 2, +0xad8
-ldxdw r1, [r7+112]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
-jne r0, 2, +0xad8
-ldxdw r1, [r7+120]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
-jne r0, 2, +0xad8
-jeq r8, 0, +0x3870
-ldxdw r1, [r7+48]
-call solana_account_info::AccountInfo::lamports#aa02
-mov64 r6, r0
-ja +0x3898
-ldxdw r1, [r7+40]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#3cfe
-jne r0, 2, +0xad8
-ldxdw r1, [r7+40]
-ldxdw r6, [r1+104]
-ldxdw r4, [r10-3176]
-ldxdw r2, [r10-3184]
-jlt r6, r2, +0x4370
-ldxdw r1, [r7+32]
-ldxdw r1, [r1+104]
-jlt r4, r1, +0x4388
-sub64 r6, r2
-sub64 r4, r1
-ldxdw r2, [r7+16]
-ldxdw r1, [r7+8]
-ldxdw r3, [r7+104]
-ldxdw r3, [r3]
-ldxdw r5, [r3+24]
-stxdw [r10-1704], r5
-ldxdw r5, [r3+16]
-stxdw [r10-1712], r5
-ldxdw r5, [r3+8]
-stxdw [r10-1720], r5
-ldxdw r3, [r3]
-stxdw [r10-1728], r3
-add64 r1, 8
-add64 r2, 8
-mov64 r5, r10
-add64 r5, -1728
-mov64 r3, r6
-stxdw [r10-3176], r4
-call orig_fn_2ce68
-jne r0, 2, +0xad8
-jeq r8, 0, +0x3a60
-ldxdw r1, [r7+48]
-call solana_account_info::AccountInfo::lamports#aa02
-ldxdw r1, [r7+8]
-ldxdw r8, [r1+80]
-ldxdw r5, [r1+72]
-ldxdw r1, [r7+16]
-ldxdw r3, [r1+48]
-ldxdw r4, [r1+40]
-ldxdw r2, [r7+120]
-ldxdw r2, [r2+104]
-stdw [r10-208], 0
-stdw [r10-216], 0
-stdw [r10-224], 0
-stdw [r10-232], 0
-stxdw [r10-1792], r5
-stxdw [r10-1776], r5
-stxdw [r10-1800], r8
-stxdw [r10-1784], r8
-stxdw [r10-1824], r6
-stxdw [r10-1736], r0
-stxdw [r10-1832], r0
-stxdw [r10-1768], r4
-stxdw [r10-1808], r4
-stxdw [r10-1760], r3
-stxdw [r10-1816], r3
-stxdw [r10-1744], r2
-ja +0x3b68
-call abort
-ldxdw r1, [r7+8]
-ldxdw r2, [r1+96]
-stxdw [r10-1784], r2
-ldxdw r1, [r1+88]
-stxdw [r10-1776], r1
-ldxdw r1, [r7+120]
-ldxdw r1, [r1+104]
-stxdw [r10-1744], r1
-ldxdw r1, [r7+40]
-ldxdw r1, [r1+104]
-stxdw [r10-1736], r1
-ldxdw r1, [r7+16]
-ldxdw r2, [r1+48]
-stxdw [r10-1760], r2
-ldxdw r2, [r1+40]
-stxdw [r10-1768], r2
-ldxdw r2, [r7+104]
-ldxdw r2, [r2]
-ldxdw r3, [r2+24]
-stxdw [r10-208], r3
-ldxdw r3, [r2+16]
-stxdw [r10-216], r3
-ldxdw r3, [r2+8]
-stxdw [r10-224], r3
-ldxdw r2, [r2]
-stxdw [r10-232], r2
-mov64 r2, 0
-stxdw [r10-1792], r2
-stxdw [r10-1800], r2
-stxdw [r10-1824], r2
-stxdw [r10-1832], r2
-stxdw [r10-1808], r2
-stxdw [r10-1816], r2
-stxdw [r10-1840], r2
-ldxdw r2, [r7+48]
-ldxdw r2, [r2]
-ldxdw r3, [r2+24]
-stxdw [r10-176], r3
-ldxdw r3, [r2+16]
-stxdw [r10-184], r3
-ldxdw r3, [r2+8]
-stxdw [r10-192], r3
-ldxdw r2, [r2]
-stxdw [r10-200], r2
-ldxdw r2, [r7+24]
 ldxdw r2, [r2+88]
 ldxdw r2, [r2]
 ldxdw r3, [r2+24]
-stxdw [r10-24], r3
+stxdw [r10-32], r3
 ldxdw r3, [r2+16]
-stxdw [r10-32], r3
+stxdw [r10-40], r3
 ldxdw r3, [r2+8]
-stxdw [r10-40], r3
+stxdw [r10-48], r3
 ldxdw r2, [r2]
-stxdw [r10-48], r2
+stxdw [r10-56], r2
+stxdw [r10-2736], r0
+jeq r7, 0, +0x2428
+call solana_account_info::AccountInfo::lamports#aa02
+stxdw [r10-2776], r0
+ldxdw r1, [r9+8]
+ldxdw r2, [r1+80]
+stxdw [r10-2744], r2
+ldxdw r1, [r1+72]
+stxdw [r10-2768], r1
+ldxdw r1, [r9+16]
+ldxdw r2, [r1+48]
+stxdw [r10-2800], r2
+ldxdw r2, [r1+40]
+stxdw [r10-2784], r2
+ldxdw r2, [r9+120]
+ldxdw r2, [r2+104]
+stxdw [r10-2816], r2
+stxdw [r10-2808], r8
+ja +0x2478
+mov64 r1, 0
+stxdw [r10-2808], r1
+ldxdw r1, [r9+16]
+mov64 r2, 0
+stxdw [r10-2776], r2
+stxdw [r10-2784], r2
+stxdw [r10-2800], r2
+stxdw [r10-2816], r2
+stxdw [r10-2768], r2
+stxdw [r10-2744], r2
 ldxdw r2, [r1+64]
-stxdw [r10-1848], r2
-ldxdw r2, [r1+56]
-stxdw [r10-1856], r2
-ldxdw r1, [r1+80]
-stxdw [r10-1864], r1
-ldxdw r1, [r7+32]
+stxdw [r10-2832], r2
+ldxdw r1, [r1+56]
+stxdw [r10-2824], r1
+ldxdw r1, [r9+32]
+mov64 r2, r9
+ldxdw r6, [r1+104]
+ldxdw r1, [r2+112]
 ldxdw r9, [r1+104]
-ldxdw r1, [r7+112]
-ldxdw r8, [r1+104]
-mov64 r1, r10
-add64 r1, -376
+mov64 r1, r10
+add64 r1, -136
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-376]
-jne r1, 1, +0x3cb0
-ldxw r2, [r10-368]
-ldxw r1, [r10-372]
+ldxw r1, [r10-136]
+jne r1, 1, +0x2508
+ldxw r2, [r10-128]
+ldxw r1, [r10-132]
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-ja +0xad8
-ldxdw r2, [r10-336]
-ldxdw r1, [r7+8]
-ldxw r1, [r1+248]
-ldxdw r3, [r7+112]
-ldxdw r3, [r3+104]
-ldxdw r4, [r10-208]
-stxdw [r10-1448], r4
-ldxdw r4, [r10-216]
-stxdw [r10-1456], r4
-ldxdw r4, [r10-224]
-stxdw [r10-1464], r4
-ldxdw r4, [r10-232]
-stxdw [r10-1472], r4
-ldxdw r4, [r10-200]
-stxdw [r10-1728], r4
-ldxdw r4, [r10-192]
-stxdw [r10-1720], r4
-ldxdw r4, [r10-184]
-stxdw [r10-1712], r4
-ldxdw r4, [r10-176]
-stxdw [r10-1704], r4
-stb [r10-1434], 1
-ldxdw r4, [r10-1824]
-stxdw [r10-1664], r4
-ldxdw r4, [r10-3176]
-stxdw [r10-1656], r4
-ldxdw r4, [r10-1832]
-stxdw [r10-1648], r4
-stxdw [r10-1640], r9
-stxdw [r10-1592], r8
-stxdw [r10-1568], r2
-ldxdw r2, [r10-1864]
-stxdw [r10-1560], r2
-ldxdw r2, [r10-1840]
-stxdw [r10-1600], r2
-stxdw [r10-1552], r2
-stxdw [r10-1544], r3
-ldxdw r2, [r10-1752]
-stxh [r10-1436], r2
-stxw [r10-1440], r1
-ldxdw r1, [r10-1792]
-stxdw [r10-1584], r1
-ldxdw r1, [r10-1800]
-stxdw [r10-1576], r1
-ldxdw r1, [r10-1856]
-stxdw [r10-1616], r1
-ldxdw r1, [r10-1848]
-stxdw [r10-1608], r1
-ldxdw r1, [r10-1808]
-stxdw [r10-1632], r1
-ldxdw r1, [r10-1816]
-stxdw [r10-1624], r1
-ldxdw r1, [r10-24]
-stxdw [r10-1672], r1
-ldxdw r1, [r10-32]
-stxdw [r10-1680], r1
-ldxdw r1, [r10-40]
-stxdw [r10-1688], r1
-ldxdw r1, [r10-48]
-stxdw [r10-1696], r1
-ldxdw r1, [r10-1784]
-stxdw [r10-1488], r1
-ldxdw r1, [r10-1776]
-stxdw [r10-1496], r1
-ldxdw r1, [r10-1760]
-stxdw [r10-1512], r1
-ldxdw r1, [r10-1768]
-stxdw [r10-1520], r1
-ldxdw r1, [r10-1744]
-stxdw [r10-1480], r1
-stxdw [r10-1504], r1
-ldxdw r1, [r10-1736]
-stxdw [r10-1528], r1
-stxdw [r10-1536], r6
-mov64 r1, r10
-add64 r1, -376
+ja +0x2cd8
+mov64 r1, 72
+ldxdw r2, [r10-2752]
+jne r2, 0, +0x2528
+mov64 r1, 88
+stxdw [r10-2840], r6
+ldxdw r3, [r10-2792]
+ldxdw r0, [r3+16]
+ldxdw r5, [r3+8]
+mov64 r2, r5
+add64 r2, r1
+ldxdw r4, [r3+112]
+ldxdw r1, [r3+120]
+ldxdw r1, [r1+104]
+mov64 r3, r1
+jne r7, 0, +0x2588
+mov64 r3, 0
+ldxdw r6, [r10-96]
+stxdw [r10-2872], r6
+ldxdw r6, [r0+48]
+stxdw [r10-2752], r6
+ldxdw r6, [r0+40]
+ldxw r5, [r5+224]
+stxdw [r10-2864], r5
+ldxdw r5, [r4+104]
+ldxdw r0, [r0+80]
+ldxdw r4, [r2+8]
+stxdw [r10-2856], r4
+ldxdw r2, [r2]
+stxdw [r10-2848], r2
+jeq r7, 0, +0x2628
+stdw [r10-112], 0
+stdw [r10-120], 0
+stdw [r10-128], 0
+stdw [r10-136], 0
+ldxdw r7, [r10-2792]
+ja +0x2680
+ldxdw r7, [r10-2792]
+ldxdw r2, [r7+104]
+ldxdw r2, [r2]
+ldxdw r4, [r2+24]
+stxdw [r10-112], r4
+ldxdw r4, [r2+16]
+stxdw [r10-120], r4
+ldxdw r4, [r2+8]
+stxdw [r10-128], r4
+ldxdw r2, [r2]
+stxdw [r10-136], r2
+ldxdw r4, [r10-2840]
+ldxdw r2, [r10-600]
+stxdw [r10-552], r2
+ldxdw r2, [r10-592]
+stxdw [r10-544], r2
+ldxdw r2, [r10-584]
+stxdw [r10-536], r2
+ldxdw r2, [r10-576]
+stxdw [r10-528], r2
+ldxdw r2, [r10-56]
+stxdw [r10-520], r2
+ldxdw r2, [r10-48]
+stxdw [r10-512], r2
+ldxdw r2, [r10-40]
+stxdw [r10-504], r2
+ldxdw r2, [r10-32]
+stxdw [r10-496], r2
+stxdw [r10-352], r4
+ldxdw r2, [r10-2776]
+stxdw [r10-360], r2
+ldxdw r2, [r10-2944]
+stxdw [r10-368], r2
+ldxdw r2, [r10-2808]
+stxdw [r10-376], r2
+stb [r10-258], 1
+ldxdw r2, [r10-2832]
+stxdw [r10-432], r2
+ldxdw r2, [r10-2824]
+stxdw [r10-440], r2
+ldxdw r2, [r10-2800]
+stxdw [r10-448], r2
+ldxdw r2, [r10-2784]
+stxdw [r10-456], r2
+stxdw [r10-336], r9
+ldxdw r2, [r10-2816]
+stxdw [r10-344], r2
+ldxdw r2, [r10-2872]
+stxdw [r10-328], r2
+stxdw [r10-320], r0
+stxdw [r10-312], r3
+stxdw [r10-304], r5
+ldxdw r2, [r10-2760]
+stxh [r10-260], r2
+ldxdw r2, [r10-2864]
+stxw [r10-264], r2
+ldxdw r2, [r10-2768]
+stxdw [r10-424], r2
+ldxdw r2, [r10-2744]
+stxdw [r10-416], r2
+stxdw [r10-296], r8
+ldxdw r2, [r10-2736]
+stxdw [r10-288], r2
+stxdw [r10-408], r6
+ldxdw r2, [r10-2752]
+stxdw [r10-400], r2
+stxdw [r10-280], r1
+ldxdw r2, [r10-2856]
+stxdw [r10-384], r2
+ldxdw r2, [r10-2848]
+stxdw [r10-392], r2
+stxdw [r10-272], r1
+ldxdw r1, [r10-112]
+stxdw [r10-464], r1
+ldxdw r1, [r10-120]
+stxdw [r10-472], r1
+ldxdw r1, [r10-128]
+stxdw [r10-480], r1
+ldxdw r1, [r10-136]
+stxdw [r10-488], r1
+mov64 r1, r10
+add64 r1, -136
 mov64 r6, r10
-add64 r6, -1728
+add64 r6, -552
 mov64 r2, r6
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#bf1e
-ldxdw r1, [r10-360]
-stxdw [r10-40], r1
-ldxdw r1, [r10-368]
+ldxdw r1, [r10-120]
 stxdw [r10-48], r1
-mov64 r1, r10
-add64 r1, -48
+ldxdw r1, [r10-128]
+stxdw [r10-56], r1
+mov64 r1, r10
+add64 r1, -56
 mov64 r8, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
 ldxdw r2, [r7+192]
 mov64 r1, r10
-add64 r1, -168
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-mov64 r1, r10
-add64 r1, -120
+add64 r1, -256
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r10
+add64 r1, -208
 mov64 r2, r6
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#bf1e
 lddw r1, .rodata[e445a52e51cb9a1d]
 lddw r2, .rodata[e445a52e51cb9a1d]
 add64 r2, 8
-stxdw [r10-56], r2
-stxdw [r10-64], r1
-ldxdw r1, [r10-120]
-stxdw [r10-80], r1
-ldxdw r1, [r10-112]
-stxdw [r10-88], r1
-stxdw [r10-96], r1
-ldxdw r2, [r10-104]
+stxdw [r10-144], r2
+stxdw [r10-152], r1
+ldxdw r1, [r10-208]
+stxdw [r10-168], r1
+ldxdw r1, [r10-200]
+stxdw [r10-176], r1
+stxdw [r10-184], r1
+ldxdw r2, [r10-192]
 add64 r1, r2
-stxdw [r10-72], r1
-mov64 r1, r10
-add64 r1, -200
+stxdw [r10-160], r1
+mov64 r1, r10
+add64 r1, -600
 mov64 r2, r10
-add64 r2, -96
+add64 r2, -184
 call <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter#0e3b
-ldxdw r6, [r10-192]
-ldxdw r7, [r10-184]
+ldxdw r6, [r10-592]
+ldxdw r7, [r10-584]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
 mov64 r3, 0
 mov64 r1, r2
 add64 r1, -34
-jle r1, r2, +0x42a8
-jeq r8, 0, +0x42b8
+jgt r1, r2, +0x2ab0
+mov64 r8, 0
+jeq r8, 0, +0x2ab8
 lddw r1, 0x300007fde
-jne r2, 0, +0x42d8
+jeq r2, 0, +0x2ad8
+mov64 r1, r3
 lddw r2, 0x300000008
-jlt r1, r2, +0x42f8
+jge r1, r2, +0x2af0
+mov64 r1, 1
+mov64 r2, 34
+call alloc::alloc::handle_alloc_error#decd
+jne r8, 0, +0x2a60
+mov64 r3, r1
+lddw r1, 0x300007fde
+jne r2, 0, +0x2a78
+lddw r2, 0x300000008
+jlt r1, r2, +0x2a98
 lddw r2, 0x300000000
 stxdw [r2], r1
-ldxdw r2, [r10-168]
+ldxdw r2, [r10-256]
 ldxdw r3, [r2]
 ldxdw r4, [r2+8]
 ldxdw r5, [r2+16]
@@ -2061,22 +1340,22 @@
 stxdw [r1+8], r4
 stxdw [r1], r3
 sth [r1+32], 1
-stxdw [r10-40], r1
-stdw [r10-32], 1
-stdw [r10-48], 1
+stxdw [r10-48], r1
+stdw [r10-40], 1
+stdw [r10-56], 1
 mov64 r8, r10
-add64 r8, -376
+add64 r8, -136
 mov64 r5, r10
-add64 r5, -48
+add64 r5, -56
 mov64 r1, r8
 lddw r2, .rodata[052ae5d7a7daa724]
 mov64 r3, r6
 mov64 r4, r7
 call solana_instruction::Instruction::new_with_bytes#33e2
 mov64 r6, r10
-add64 r6, -48
+add64 r6, -56
 mov64 r2, r10
-add64 r2, -168
+add64 r2, -256
 mov64 r1, r6
 mov64 r3, 48
 call memcpy
@@ -2090,40 +1369,79 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x4310
-mov64 r6, 2
-ja +0x4330
-mov64 r3, r10
-add64 r3, -382
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
-call core::result::unwrap_failed#5a05
-mov64 r8, 0
-jne r8, 0, +0x40b0
-mov64 r3, r1
-lddw r1, 0x300007fde
-jeq r2, 0, +0x40c8
-mov64 r1, r3
-lddw r2, 0x300000008
-jge r1, r2, +0x40e0
-mov64 r1, 1
-mov64 r2, 34
-call alloc::alloc::handle_alloc_error#decd
+jne r1, 26, +0x2c68
+mov64 r7, 2
+ja +0x2c88
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-mov64 r6, r0
-mov64 r7, r1
-mov64 r1, r10
-add64 r1, -48
+mov64 r7, r0
+mov64 r6, r1
+mov64 r1, r10
+add64 r1, -56
 call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 1]>#8f15
-mov64 r0, r6
-mov64 r1, r7
-jne r0, 2, +0xad8
+jne r7, 2, +0x2ce8
+mov64 r1, r10
+add64 r1, -2688
+call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 23]>#60d1
 mov64 r0, 2
-ja +0xad8
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:595:23)
+ja +0x620
+call mayhem_program::instructions::pump_interface::buy::math#3277
+mov64 r7, r0
+mov64 r6, r1
+mov64 r1, r10
+add64 r1, -2688
+call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 23]>#60d1
+mov64 r1, r6
+mov64 r0, r7
+ja +0x620
+ldxdw r1, [r10-2684]
+stxdw [r10-552], r1
+mov64 r3, r10
+add64 r3, -552
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:295:23)
+call core::result::unwrap_failed#5a05
+ldxdw r1, [r10-2684]
+stxdw [r10-552], r1
+mov64 r3, r10
+add64 r3, -552
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:301:24)
+call core::result::unwrap_failed#5a05
+mov64 r3, r10
+add64 r3, -1
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:314:54)
+call core::result::unwrap_failed#5a05
+stw [r10-2688], 11
+mov64 r3, r10
+add64 r3, -2688
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Debug>::fmt#6591]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:318:123)
+call core::result::unwrap_failed#5a05
+mov64 r3, r10
+add64 r3, -1
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:318:133)
+call core::result::unwrap_failed#5a05
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:312:15)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:596:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:318:58)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
+mov64 r1, 1
+mov64 r2, 1
+call alloc::alloc::handle_alloc_error#decd
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:330:10)
+call core::panicking::panic_const::panic_const_sub_overflow#7310
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_amm_interface.rs:330:44)
+call core::panicking::panic_const::panic_const_sub_overflow#7310
